### ISSUE TITLE
Service targetPort incorrect

### ISSUE THREAD

**Issue by endakelly** [oauth2-proxy/manifests#86]
When creating a helm release from this chart, the upstream service doesn't load at all unless the service targetPort is edited. My httpScheme value is set to https. To get the static upstream page to work, I have to edit the service and change the targetPort from https to 4443.

After this, the page loads successfully.

**Before:** 
  
ports:
  - name: https
    nodePort: 31936
    port: 443
    protocol: TCP
    targetPort: https

![image](https://user-images.githubusercontent.com/80516065/154967320-dfac8746-92e0-4db1-8cea-330cae05d8ae.png)

**After:**

  ports:
  - name: https
    nodePort: 31936
    port: 443
    protocol: TCP
    targetPort: 4443

![image](https://user-images.githubusercontent.com/80516065/154967517-ba769c61-02cc-4ed8-9ce2-410404024a13.png)

**Comment by bluebrown** [comment]
Where does this port name even come from in the first place? I cannot find it i[n the chart](https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/templates/service.yaml). What version of the cart are you using?

Service has those ports:

```yaml
  ports:
    - port: {{ .Values.service.portNumber }}
      targetPort: {{ .Values.httpScheme }}
      protocol: TCP
      name: {{ .Values.httpScheme }}
    {{- if and .Values.metrics.enabled .Values.metrics.port }}
    - port: {{ .Values.metrics.port }}
      protocol: TCP
      targetPort: metrics
      name: metrics
    {{- end }}
```

The value for `httpScheme` is http, and it points to `service.portNumber` which is 80 by default.

**Comment by pierluigilenoci** [comment]
@endakelly feel free to open a PR.

**Comment by endakelly** [comment]
@pierluigilenoci I tried to push a new branch and create a PR but I don't seem to have permission.

@bluebrown further to my initial comments, in my config the https-address is set to ":4443". If I change this to ":443" then I get a CrashLoopBackOff error when starting the pod. The container port remains 4180 and I haven't had a need to alter it.

**Comment by bluebrown** [comment]
@endakelly you need to fork the repo and create the branch there and make the PR from there. 

Regarding the container port. Target is effectively the container port.  I still don't really understand your setup. How can you have the container port set to 4180 but have it working when the service targets 4443? This port should not be exposed at all, even though you may have configured it in your config.

Apart from that, if this port is set in your config, It's nothing the chart could anticipate really. What if tomorrow someone sets it to 1234? 

Unless you find a way to read that value from the config, and use it in the chart for all use cases, making a correct change may be hard.

I guess the most easy solution would be to set the port in your config to 4180 for https and then set httpScheme to https.

**Comment by bluebrown** [comment]
I am talking about this https://github.com/oauth2-proxy/manifests/blob/03ad9c3dfa6ccf52557a60ac9261a30afcf15a6e/helm/oauth2-proxy/templates/deployment.yaml#L150

The name for that port will be whatever you write in your scheme. But no other port is used, regardless what you have in your config.

Can you show you config? How did you set the port to 4443?

**Comment by endakelly** [comment]
PR submitted https://github.com/oauth2-proxy/manifests/pull/89

Here is the bulk of my config: https://github.com/hmcts/sds-flux-config/blob/master/k8s/namespaces/admin/oauth2-proxy/oauth2-proxy.yaml

Here is some of the config I'm using: https://github.com/hmcts/sds-flux-config/blob/master/k8s/namespaces/admin/oauth2-proxy/oauth2-proxy.yaml
There's some missing args from that config currently. I've got more args locally in my test branch e.g. upstream, redirect-url

**Comment by bluebrown** [comment]
How does this change the hard coded port 4180? I can't imagine this works. If you just change the target port to something and kubernetes is not publishing this port on the container, it cant work.

**Comment by endakelly** [comment]
OK I see. I was updating the service targetPort to 4443 without changing the containerPort. Updating the containerPort to 4443 while keeping the service targetPort also makes the page load correctly.

I will update my PR with your suggestion in https://github.com/oauth2-proxy/manifests/issues/86#issuecomment-1047187412

**Comment by endakelly** [comment]
As to why it works when changing the service targetPort to 4443, I don't have an answer to that

**Comment by bluebrown** [comment]
IMO, both ports should be configurable. In the settings, oauth2-proxy has http and https port. The current implementation only supports http. Your PR supports overriding that. But I think both should be configurable and useable at the same time. At least it's a feature of oauth2-proxy itself to have both.

**Comment by pierluigilenoci** [comment]
Closed with #89
